src/dstack/_internal/server/routers/runs.py

```diff
@@ -58,9 +66,48 @@ async def get_run(
     return run
 
 
+@project_router.post("/get_offers")
+async def get_offers(
+    body: GetOffersRequest,
+    session: AsyncSession = Depends(get_session),
+    user_project: Tuple[UserModel, ProjectModel] = Depends(ProjectMember()),
+) -> Tuple[str, List[InstanceOfferWithAvailability]]:
```

Comment:
Define models for complex responses since working with tuples at the API level is inconvenient (e.g. hard-to-read openapi schema).

Reply:
Fixed